dd-java-agent/agent-iast/src/main/java/com/datadog/iast/sink/SinkModuleBase.java

```diff
@@ -135,14 +135,13 @@ public void registerDependencies(@Nonnull final Dependencies dependencies) {
     return result;
   }
 
-  protected final void report(
+  protected final Location report(
```

Comment:
In my opinion is a bit strange get the location from a report method, what do you think about create a new method to get the Location and overload the report to get as parameter the Location? for instance:

```suggestion
  protected final Location getLocationForSpanAndStack(final AgentSpan span){
  return Location.forSpanAndStack(spanId(span), getCurrentStackTrace());
  }
  
  protected final Location report(@Nullable final AgentSpan span,
      @Nonnull final VulnerabilityType type,
      @Nonnull final Evidence evidence,
      @Nonnull final Location location) {
    reporter.report(span, new Vulnerability(type, location, evidence));
  }
  
  protected final Location report(@Nullable final AgentSpan span,
      @Nonnull final VulnerabilityType type,
      @Nonnull final Evidence evidence) {
    reporter.report(span, new Vulnerability(type, Location.forSpanAndStack(spanId(span), getCurrentStackTrace()), evidence));
  }
```

Reply:
this code wasn't even needed, it was merged from the previous PR. My mistake. I have removed it.